cli/Sources/TuistGenerator/Mappers/ResourcesProjectMapper.swift

```diff
@@ -376,6 +395,33 @@ public class ResourcesProjectMapper: ProjectMapping { // swiftlint:disable:this
     }
 }
 
+extension BuildableFolder {
+    fileprivate var containsSourceLikeFiles: Bool {
+        if hasExtension(path.extension, in: Target.validSourceCompatibleFolderExtensions) {
+            return true
+        }
+        return resolvedFiles.contains { file in
+            hasExtension(file.path.extension, in: Target.validSourceExtensions) ||
+                hasExtension(file.path.extension, in: Target.validSourceCompatibleFolderExtensions)
+        }
+    }
+
+    fileprivate var containsResourceLikeFiles: Bool {
+        if hasExtension(path.extension, in: Target.validResourceCompatibleFolderExtensions) {
+            return true
+        }
+        return resolvedFiles.contains { file in
+            hasExtension(file.path.extension, in: Target.validResourceExtensions) ||
+                hasExtension(file.path.extension, in: Target.validResourceCompatibleFolderExtensions)
+        }
+    }
+}
+
+private func hasExtension(_ extensionValue: String?, in allowedExtensions: [String]) -> Bool {
```

Comment:
this can be an extension on `AbsolutePath`, should make the code a bit prettier ✨